irods/session.py

```diff
@@ -355,10 +376,37 @@ def port(self):
 
     @property
     def server_version(self):
+        return self._server_version()
+
+    def server_version_without_auth(self):
+        """Returns the same version tuple as iRODSSession's server_version property, but
+        does not require successful authentication.
+        """
+        from irods.connection import Connection
+
+        with self.clone().pool.no_auto_authenticate() as pool:
+            return Connection(pool, pool.account).server_version
+
+    GET_SERVER_VERSION_WITHOUT_AUTH = staticmethod(
+        lambda s: s.server_version_without_auth()
+    )
+
+    def _server_version(self, version_func=None):
+        """The  server version can be retrieved by the usage
+            session._server_version()
+        with conditional substitution by another version by use of the environment variable:
```

Comment:
Add empty line around L396 so the function stands out more.

Reply:
done